src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -416,6 +417,12 @@ public void messageReceived(RecoveryCleanFilesRequest request, TransportChannel
 
     class FileChunkTransportRequestHandler extends BaseTransportRequestHandler<RecoveryFileChunkRequest> {
 
+        // When we last consulted RateLimiter:
+        long lastPauseCheckBytes;
```

Comment:
can we make these atomic longs? this is a message handler that can (in theory) be called concurrently